### PR TITLE
PS-4996, PS-3956: Fixing user statistics bugs (5.6)

### PR DESCRIPTION
**Pull request by dutow**

### CODE REVIEW

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
If this is a backport cherrypick, use -x option to include original commit hash, and copy its commit message.
If this is not a cherrypick, please expand the commit message what and how is being fixed

**Comment by dutow** [comment]
Commit message edited. In 5.6, I've added tests to every variable which was mentioned in the tickets, but only 2 weren't working. In 5.7 / 8.0, a few more fixes had to be added - I'll open those pull requests based on this after this is approved.

File: sql/handler.cc [hunk 1]
```diff
@@ -1530,6 +1530,10 @@ int ha_commit_trans(THD *thd, bool all, bool ignore_global_read_lock)
   /* Free resources and perform other cleanup even for 'empty' transactions. */
   if (is_real_trans)
     thd->transaction.cleanup();
+
+  if(!error)
```
Comment (laurynas-biveinis) [inline sql/handler.cc]: Space after if

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
Typo in the commit message: "statisctics".

OK with both comments fixed

**Comment by dutow** [comment]
https://ps.cd.percona.com/job/percona-server-5.6-param/44/

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**